### ISSUE TITLE
pytorch 0.3 compatibility?

### ISSUE THREAD

**Issue by rainjacket** [NVIDIA/sentiment-discovery#14]
To what degree is the code backwards compatible with pytorch 0.3? I can't easily compile pytorch from source in my current environment, inside a prebuilt docker image.

It seems like if not using dynamic_loss_scale, I can at least train and get a lang_model.pt file with 0.3, is that right?

**Comment by raulpuric** [comment]
That should be correct. It may or may not work with distributed nccl/fp16 training though.

0.3 will fail on eval, classification, transfer, generation however as it does not have the `with torch.no_grad()` functionality included in >0.4

**Comment by rainjacket** [comment]
Hijacking this thread to ask a separate question (because I'm not sure if there's an actual issue). For me, it appears that the model training makes no progress when using the default optimizer, only when setting --optim=Adam does the loss decrease over time. Am I doing something wrong?

And thanks for the help @raulpuric and for releasing this project 👍

**Comment by raulpuric** [comment]
Yeah sorry Adam really should be the default.
With SGD one has to use a much higher LR.
